### PR TITLE
Trial moves

### PR DESCRIPTION
**Pull request by chrisjonesBSU**
This PR finishes the `trial_moves` and `run` functions as well as adding a couple new attributes to the `Simulation` class.

### CODE REVIEW

File: MCMC/simulation.py [hunk 1]
```diff
@@ -53,18 +59,46 @@ def trial_move(self):
...
+        if delta_U <= 0:
+            self.system = _system
+            self.accepted_moves += 1
+        else:
+            rand_num = random.uniform(0, 1)
+            if np.exp(-delta_U/self.kT) <= rand_num:
+                self.system = _system
+                self.accepted_moves += 1
+            else:
+                pass
```
Comment (marjanAlbouye) [inline MCMC/simulation.py:87]: I'd suggest moving accept/reject section to a new function that takes delta_U as an argument. We might develop multiple versions of trial moves but the acceptance criteria will remain the same for all.

File: MCMC/simulation.py [hunk 2]
```diff
@@ -53,18 +59,46 @@ def trial_move(self):
...
         """
-        return NotImplementedError
+        _system = np.copy(self.system)
+        # Select random occupied site
+        non_zero_indices = list(np.nonzero(self.system))
+        random.shuffle(non_zero_indices)
+        site1 = non_zero_indices[0]
+        site1_val = _system[site1]
+        # Select random site from all possible sites
+        all_indices = list(self.grid_indices)
```
Comment (marjanAlbouye) [inline MCMC/simulation.py:69]: Do we wanna exclude `site1` from this list of potential candidates?

**Review by marjanAlbouye — APPROVED** [review]
Looks good. Thanks Chris! Just left a few notes.

File: MCMC/simulation.py (hunk 2 above)
Comment (chrisjonesBSU) [reply MCMC/simulation.py:69]: I thought about that, but not sure if that is considered a valid move and we want to keep it as a possible out come (i.e. nothing changes during this step).

**Comment by chrisjonesBSU** [comment]
Switching the a non-lattice system.

**Pull request closed without merging**